oci/table_oci_devops_project.go

```diff
@@ -0,0 +1,314 @@
+package oci
+
+import (
+	"context"
+	"strings"
+
+	oci_common "github.com/oracle/oci-go-sdk/v65/common"
+	"github.com/oracle/oci-go-sdk/v65/devops"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/turbot/go-kit/types"
+
+	"github.com/turbot/steampipe-plugin-sdk/v4/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableDevopsProject(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:             "oci_devops_project",
+		Description:      "OCI Devops Project",
+		DefaultTransform: transform.FromCamel(),
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.SingleColumn("id"),
+			Hydrate:    getProject,
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listProjects,
+			KeyColumns: []*plugin.KeyColumn{
+				{
+					Name:    "compartment_id",
+					Require: plugin.Optional,
+				},
+				{
+					Name:    "name",
+					Require: plugin.Optional,
+				},
+				{
+					Name:    "id",
+					Require: plugin.Optional,
+				},
```

Comment:
Would it be possible to add `lifecycle_state` as an optional `KeyColumn`

Refer - https://pkg.go.dev/github.com/oracle/oci-go-sdk/v65@v65.28.0/devops#ListProjectsRequest